app/javascript/blacklight/modalForm.js

```diff
@@ -0,0 +1,60 @@
+// The email and sms forms are displayed inside a modal.  When the form is submitted,
+// this script closes the modal and puts the output on the main part of the page.
+// By default it is rendering catalog/sms_success and catalog/email_succcess.
+// These templates deliver a payload of the format used by turbo-streams.
+// See https://turbo.hotwired.dev/handbook/streams
+// That format allows a downstream application to override the template to define
+// multiple customizable areas of the page to get updated.
+export default class {
+    constructor(errorHandler, badRequestHandler, hideModal) {
+      this.errorHandler = errorHandler
+      this.badRequestHandler = badRequestHandler
+      this.hideModal = hideModal
+    }
+
+    get triggerFormSelector() {
+      return 'form[data-blacklight-modal~=trigger]'
+    }
+
+    bind() {
+      document.addEventListener('submit', (e) => {
+         if (e.target.matches(this.triggerFormSelector))
+          this.onSubmit(e)
+      })
+    }
+
+    // This is like a light-weight version of turbo that only supports append presently.
+    updateTurboStream(data) {
```

Comment:
Question: This isn't actually turbo streams, but something that uses similar naming conventions, yeah? Is this just prepping Blacklight to head in the direction of turbo streams?